src/Pact/Types/Runtime.hs

```diff
@@ -422,8 +422,8 @@ newtype EnvReadOnly e = EnvReadOnly (EvalEnv e)
 instance PureReadOnly (EnvReadOnly e)
 instance PureNoDb (EnvReadOnly e)
 
-diePure :: Method e a
-diePure _ = throwM $ PactError EvalError def def "Illegal database access in pure context"
+diePure :: Text -> Method e a
+diePure opName _ = throwM $ PactError EvalError def def $ "Illegal access attempt (" <> pretty opName <> ")"
```

Comment:
I prefer leaving "database" in the error message. It's probably also worth leaving in something like "in a context which only allows systable reads".

Reply:
I left out "database" because there are some operations that are disallowed via `diePure` that an end user probably wouldn't associate with this word; e.g. "begin-tx". But I can re-add that word.

We can't say "in a context which only allows systable reads" because `diePure` is not only run in pure mode. It's also used in read-only mode. We should probably improve the name `diePure`.